runtime/parser/type.go

```diff
@@ -544,10 +545,36 @@ func defineRestrictedOrDictionaryType() {
 	)
 }
 
-// parseNominalTypes parses zero or more nominal types separated by comma.
+func parseNominalType(
+	p *parser,
+	rightBindingPower int,
+	rejectAccessKeywords bool,
```

Comment:
Maybe instead of adding specific "access keyword rejection" support to the nominal type parsing here and to `parseNominalTypes`, maybe just parse all identifiers as nominal types, and then reject nominal type lists that have nominal types which are e.g. access keywords in the case of parsing entitlements.

Also/alternatively, given that this feature goes into Stable Cadence: Naomi had worked on rejecting identifiers which are also keywords in other places, see e.g. `nonReservedIdentifier`, so maybe we can just use that here and generalize from specifically rejecting access-related keywords to all hard keywords.

Reply:
Right, I just meant that this rejection functionality is currently "pushed down" all the way into the nominal type parsing function(s), though the callers of those parsing functions can handle it themselves: if they parse `false` for `rejectAccessKeywords`, then no rejection is needed; if they pass `true`, then the callr can also just reject the result of the call.

Reply:
Extracted this out into a wrapper function.